File: swift/difference-of-squares/Sources/DifferenceOfSquares/DifferenceOfSquares.swift
//Solution goes in Sources

struct Squares {
    var number: Int
    
    init(_ number: Int){
        self.number = number
    }
    
    var squareOfSum: Int {
        var sum: Int = 0
        for num in 1...number {
            sum += num
        }
        
        return sum*sum
    }
    
    var sumOfSquares: Int {
        var sum: Int = 0
        for num in 1...number {
            sum += num*num
        }
        
        return sum
    }
    
    var differenceOfSquares: Int {
        squareOfSum - sumOfSquares
    }
}

File: go/difference-of-squares/difference_of_squares.go
package diffsquares

func SquareOfSum(n int) int {
	var value int
	for i := 0; i <= n; i++ {
		value += i
	}

	return value * value
}

func SumOfSquares(n int) int {
	var value int
	for i := 0; i <= n; i++ {
		value += i * i
	}

	return value
}

func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}
